test/integration/integration_admin_test.cc

```diff
@@ -448,6 +453,41 @@ TEST_P(IntegrationAdminTest, Admin) {
   envoy::admin::v2alpha::SecretsConfigDump secret_config_dump;
   config_dump.configs(5).UnpackTo(&secret_config_dump);
   EXPECT_EQ("secret_static_0", secret_config_dump.static_secrets(0).name());
+
+  // Validate that the inboundonly is processed correctly.
+  response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "POST",
+                                                "/drain_listeners?inboundonly", "",
+                                                downstreamProtocol(), version_);
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().Status()->value().getStringView());
+  EXPECT_EQ("text/plain; charset=UTF-8", ContentType(response));
+  EXPECT_EQ("OK\n", response->body());
+
+  // Validate that the drain_listeners is processed correctly.
+  response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "POST", "/drain_listeners", "",
+                                                downstreamProtocol(), version_);
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().Status()->value().getStringView());
+  EXPECT_EQ("text/plain; charset=UTF-8", ContentType(response));
+  EXPECT_EQ("OK\n", response->body());
+}
+
+// Validates that the inboundonly drains inbound listeners.
```

Comment:
How is this actually validated? Is there a stat you can wait on, etc? I'm not sure how this test is different in practice from the one above unless you verify that the drain actually happens via some mechanism?

Reply:
There is no stat to validate- it just validates that call is done with out errors when we pass the additional argument. I will correct the comment though

Reply:
Why can't you validate that draining/deletion actually happens? I'm not sure the point of this test otherwise? How is this any different from the above tests?

Reply:
Oh. Sorry, This test is not required. I will remove this.

Reply:
Also enhanced it to cover few more cases. We should be good.